### ISSUE TITLE
Error: axis() got an unexpected keyword argument 'y_min' 

### ISSUE THREAD

**Issue by sabarezaeii** [rasbt/mlxtend#1017]
Hello,
I am having the same issue.
I am trying to run a very simple example to check plot_decision_regions and I get the following error:

axis() got an unexpected keyword argument 'y_min'

my code:

X = np.array([[0.3,1],[0.5,4.5],[1,2.3],[1.4,1.9],[1.7,8.9],[2,4.1],[3.3,7],[3.5,1.5],[4,6.3],[4.4,1.9],[5.7,2.9],[6,7.1]])
y = np.array([0,0,0,0,0,0,1,1,1,1,1,1])
plot_decision_regions( X, y, clf=classifier, legend=2)

plt.show()

**Comment by sabarezaeii** [comment]
I also have tried this below code which I found in in your github http://rasbt.github.io/mlxtend/user_guide/plotting/plot_decision_regions/
and ran that both on a jupyter notebook on my computer and google Collab, and I gat the same error: axis() got an unexpected keyword argument 'y_min' !!


from mlxtend.plotting import plot_decision_regions
import matplotlib.pyplot as plt
from sklearn import datasets
from sklearn.svm import SVC

# Loading some example data
iris = datasets.load_iris()
X = iris.data[:, [0, 2]]
y = iris.target

# Training a classifier
svm = SVC(C=0.5, kernel='linear')
svm.fit(X, y)


# Plotting decision regions
plot_decision_regions(X, y, clf=svm, legend=2)

# Adding axes annotations
plt.xlabel('sepal length [cm]')
plt.ylabel('petal length [cm]')
plt.title('SVM on Iris')
plt.show()

**Comment by rasbt** [comment]
Hm , that's weird. Haven't seen that before. What version of mlxtend are you using? Can you do an

```
import mlxtend
print(mlxtend.__version__)
```

**Comment by sabarezaeii** [comment]
Thanks for responding.

I have tried two different versions of mlxtend:
0.13.0 and 0.21.0

is that possible that matplotlib has updated something that caused this issue?

**Comment by rasbt** [comment]
Good point. Haven't updated my matplotlib in a while. What version are you using? I am currently on '3.5.1' where it works ok.

**Comment by rasbt** [comment]
<img width="806" alt="Screen Shot 2023-02-23 at 3 19 19 PM" src="https://user-images.githubusercontent.com/5618407/221032599-a2b332aa-e4cd-4b9e-a481-f08d20ac3924.png">

**Comment by sabarezaeii** [comment]
yesss, That was a matplotlib version problem. now I have matplotlib 3.5.3 and mlxtend 0.21.0, and everything is working.


thank you for your help.

**Comment by rasbt** [comment]
glad to hear that it could be (relatively) easily resolved!

**Comment by rasbt** [comment]
Ah sorry, forgot to ask: did you have an older or newer matplotlib version?

**Comment by sabarezaeii** [comment]
I had matplotlib 3.5.2.

**Comment by rasbt** [comment]
oh maybe an intermediate issue between 3.5.1 and 3.5.3 then hmmm
